elasticdl_preprocessing/layers/to_sparse.py

```diff
@@ -0,0 +1,59 @@
+import tensorflow as tf
+
+
+class ToSparse(tf.keras.layers.Layer):
+    """Converts a `Tensor` to a `SparseTensor`, dropping ignore_value cells.
+    If the input is already a `SparseTensor`, just return it.
+
+    Example :
+    ```python
+        layer = ToSparse()
+        inp = tf.constant([["A", ""], ["B", "C"]], tf.string)
+        layer.call(inp)
+        tf.SparseTensor(
+            indices=np.array([[0, 0], [1, 0], [1, 1]]),
+            values=np.array(["A", "B", "C"]),
+            dense_shape=(2, 2),
+        )
+    ```
+
+    Arguments:
+        ignore_value: Entries in inputs equal to this value will be
+            absent from the output `SparseTensor`. If `None`, default value of
+            inputs dtype will be used ('' for `str`, -1 for `int`).
```

Comment:
Should we expose this?

Reply:
-1 for `int` seems dangerous as this is application specific

Reply:
-1 for `int` is the default value and the layer will ignore it during transformation.

Reply:
> -1 for `int` seems dangerous as this is application specific

Use -1 as the default ignore_value for `int` is also the implementation inside feature column. Please check the [code snippet](https://github.com/tensorflow/tensorflow/blob/983a5e245e703ba8b6dd84007fed6e641f741ab0/tensorflow/python/feature_column/feature_column_v2.py#L2769-L2801).